File: leetcode/AIcoding/3.DeepSeek-v3/java_solutions/2507-smallest-value-after-replacing-with-sum-of-prime-factors.java
class Solution {
    public int smallestValue(int n) {
        while (true) {
            int sum = getPrimeFactorsSum(n);
            if (sum == n) {
                break;
            }
            n = sum;
        }
        return n;
    }

    private int getPrimeFactorsSum(int num) {
        int sum = 0;
        int temp = num;
        for (int i = 2; i <= temp; i++) {
            while (temp % i == 0) {
                sum += i;
                temp /= i;
            }
        }
        return sum;
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/go_solutions/2507-smallest-value-after-replacing-with-sum-of-prime-factors.go
func smallestValue(n int) int {
	for {
		sum := 0
		temp := n
		for i := 2; i*i <= temp; i++ {
			for temp%i == 0 {
				sum += i
				temp /= i
			}
		}
		if temp > 1 {
			sum += temp
		}
		if sum == n {
			return n
		}
		n = sum
	}
}